tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterTestBase.java

```diff
@@ -369,32 +426,53 @@ public void testUploadingXnumberOfMessages(final TestContext ctx) throws Interru
     private Future<ProtonConnection> connectToAdapter(final String username, final String password) {
 
         final Future<ProtonConnection> result = Future.future();
-        final ProtonClientOptions options = new ProtonClientOptions();
         final ProtonClient client = ProtonClient.create(vertx);
 
-        client.connect(options, IntegrationTestSupport.AMQP_HOST, IntegrationTestSupport.AMQP_PORT,
-                username, password, conAttempt -> {
-            if (conAttempt.failed()) {
-                result.fail(conAttempt.cause());
-            } else {
-                this.context = Vertx.currentContext();
-                this.connection = conAttempt.result();
-                connection.openHandler(remoteOpen -> {
-                    if (remoteOpen.succeeded()) {
-                        result.complete(connection);
-                    } else {
-                        result.fail(remoteOpen.cause());
-                    }
-                });
-                connection.closeHandler(remoteClose -> {
-                    connection.close();
-                });
-                connection.open();
-            }
-        });
+        if (!Strings.isNullOrEmpty(username) && !Strings.isNullOrEmpty(password)) {
```

Comment:
please split this up into two methods: *connectToAdapter(String, String)* and *connectToAdapter(SelfSignedCertificate)* to make it more transparent in the client code which mechanism is used. There is basically no duplicated code so it shouldn't be an issue ...

Reply:
ok :+1: